src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -65,6 +65,78 @@ public static bool TryGetEnvironmentVariables(this IResource resource, [NotNullW
         return TryGetAnnotationsOfType(resource, out environmentVariables);
     }
 
+    /// <summary>
+    /// Get the environment variables from the given resource.
+    /// </summary>
+    /// <remarks>
+    /// This method is useful when you want to make sure the environment variables are added properly to resources, mostly in test situations.
+    /// This method has asynchronous behavior when <paramref name = "applicationOperation" /> is <see cref="DistributedApplicationOperation.Run"/>
+    /// and environment variables were provided from <see cref="IValueProvider"/> otherwise it will be synchronous.
+    /// </remarks>
+    /// <param name="resource">The resource to get the environment variables from.</param>
+    /// <param name="applicationOperation">The context in which the AppHost is being executed.</param>
+    /// <returns>The environment variables retrieved from the resource.</returns>
+    /// <example>
+    /// <code>
+    /// using var builder = await DistributedApplicationTestingBuilder.CreateAsync&lt;SampleShop_AppHost&gt;();
+    ///
+    /// builder.AddMongoDB("mongo")
+    ///     .WithEndpoint("tcp", e => e.AllocatedEndpoint = new AllocatedEndpoint(e, "localhost", 3000, containerHost))
+    ///     .WithMongoExpress();
+    ///
+    /// var env = await mongoExpress.GetEnvironmentVariableValuesAsync();
```

Comment:
Does this code build?

1. What is the `mongoExpress` variable? I don't see it defined in the sample.
2. The result of `DistributedApplicationTestingBuilder.CreateAsync` is an `IDistributedApplicationTestingBuilder` which doesn't have an `.AddMongoDB` extension method.

Reply:
Yeah, I think a more realistic scenario for this API usage is actually inside a lifecycle hook. It's hard to show that without a much more detailed sample.